tequilapi/http_api_server.go

```diff
@@ -0,0 +1,74 @@
+package tequilapi
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type ApiServer interface {
+	Port() int
+	Wait()
+	Stop()
+}
+
+type apiServer struct {
+	listener  net.Listener
+	stopped   *sync.WaitGroup
+	boundPort int
+}
+
+func NewServer(address string, port int, handler http.Handler) (ApiServer, error) {
+	var err error
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:%d", address, port))
+	if err != nil {
+		return nil, err
+	}
+
+	boundPort, err := extractBoundPort(listener)
+	if err != nil {
+		listener.Close()
+		return nil, err
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	server := apiServer{listener, &wg, boundPort}
+	go serve(listener, handler, &wg)
```

Comment:
I think if we make `Serve` a public method, then we could get rid of some of this confusing `sync.WaitGroups`

Reply:
serve method is internal implementation and is not needed to be public, it is also a blocking function that's why is started in separate go routine, and in order to find out when server finished serving requests, waitgroup synchronization is used.

Reply:
but do we really need two syncGroups here? 
My thinking is that `server.stopped` could be enough if we separate server construction and running.

Reply:
I only see one syncGroup here.